src/main/java/com/netflix/imflibrary/st2067_203/MGASADMTrackFileConstraints.java

```diff
@@ -0,0 +1,372 @@
+/*
+ *
+ * Copyright 2024 RheinMain University of Applied Sciences, Wiesbaden, Germany.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+package com.netflix.imflibrary.st2067_203;
+
+import com.netflix.imflibrary.IMFConstraints;
+import com.netflix.imflibrary.IMFErrorLogger;
+import com.netflix.imflibrary.st0377.CompoundDataTypes;
+import com.netflix.imflibrary.st0377.HeaderPartition;
+import com.netflix.imflibrary.st0377.IndexTableSegment;
+import com.netflix.imflibrary.st0377.header.AudioChannelLabelSubDescriptor;
+import com.netflix.imflibrary.st0377.header.GenericDescriptor;
+import com.netflix.imflibrary.st0377.header.GenericPackage;
+import com.netflix.imflibrary.st0377.header.GenericSoundEssenceDescriptor;
+import com.netflix.imflibrary.st0377.header.GroupOfSoundFieldGroupLabelSubDescriptor;
+import com.netflix.imflibrary.st0377.header.InterchangeObject;
+import com.netflix.imflibrary.st0377.header.Preface;
+import com.netflix.imflibrary.st0377.header.Sequence;
+import com.netflix.imflibrary.st0377.header.SoundFieldGroupLabelSubDescriptor;
+import com.netflix.imflibrary.st0377.header.SourcePackage;
+import com.netflix.imflibrary.st0377.header.TimelineTrack;
+import com.netflix.imflibrary.st0377.header.UL;
+import com.netflix.imflibrary.st0377_41.MCAContent;
+import com.netflix.imflibrary.st0377_41.MCAUseClass;
+import com.netflix.imflibrary.st379_2.ContainerConstraintsSubDescriptor;
+import com.netflix.imflibrary.utils.DOMNodeObjectModel;
+import com.netflix.imflibrary.utils.ErrorLogger;
+
+import javax.annotation.Nonnull;
+
+import static com.netflix.imflibrary.st0377.header.GenericPictureEssenceDescriptor.containerConstraintsSubDescriptorUL;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public final class MGASADMTrackFileConstraints {
+
+    private static final String IMF_MGASADM_EXCEPTION_PREFIX = "IMF MGA S-ADM check: ";
+    private static final UL SerialAudioDefinitionModelMetadataPayload = UL.fromULAsURNStringToUL("urn:smpte:ul:060e2b34.0401010d.04040212.00000000");
+
+    // Prevent instantiation
+    public MGASADMTrackFileConstraints() {}
+
+    public static void checkCompliance(IMFConstraints.HeaderPartitionIMF headerPartitionIMF, @Nonnull IMFErrorLogger imfErrorLogger) throws IOException {
```

Comment:
Same comment re `FATAL` errors as [here](https://github.com/Netflix/photon/pull/376/files#r1735395501).

Reply:
All FATAL errors changed to NON_FATAL.